### PR TITLE
Problem: don't support debug query result (backport: #328)

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- update cosmos-sdk dependency, the feature is backported to cosmos-sdk 0.44.x

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -157,11 +157,11 @@ replace github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1
...
 replace github.com/cosmos/ibc-go => github.com/crypto-org-chain/ibc-go v1.2.1-hooks
 
-replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.1
+replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.3
 
 replace github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.3-patched
 
 // TODO: remove when ibc-go and ethermint upgrades cosmos-sdk
-replace github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.44.2
+replace github.com/cosmos/cosmos-sdk => github.com/yihuang/cosmos-sdk v0.44.6-0.20220214143659-d3f9eb1a81ea
```
Comment (yihuang) [inline go.mod:165]: update after upstream merged: https://github.com/cosmos/cosmos-sdk/pull/11181

**Comment by thomas-nguy** [comment]
why is this PR pointed to release/v0.6.x?

**Comment by yihuang** [comment]
> why is this PR pointed to release/v0.6.x?

it's a backport, the feature is backported to cosmos-sdk 0.44.x

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/armon/go-metrics v0.3.9
-	github.com/cosmos/cosmos-sdk v0.44.3
+	github.com/cosmos/cosmos-sdk v0.44.6
```
Comment (thomas-nguy) [inline go.mod:7]: https://github.com/cosmos/cosmos-sdk/compare/v0.44.3...v0.44.6

Comment (yihuang) [reply go.mod:7]: actually: https://github.com/cosmos/cosmos-sdk/compare/v0.44.3...8a26cd10b4be

**Review by thomas-nguy — APPROVED** [review]
looks fine after a quick check of the changelog of cosmos sdk

Just to confirm, events are not store in consensus right?
https://github.com/cosmos/cosmos-sdk/pull/10631

**Review by adu-crypto — APPROVED** [review]

**Comment by yihuang** [comment]
> Just to confirm, events are not store in consensus hash right? [cosmos/cosmos-sdk#10631](https://github.com/cosmos/cosmos-sdk/pull/10631)

I think it is stored in consensus hash, so it looks like breaking, maybe better to cherry-pick to a custom fork then.

**Comment by tomtau** [comment]
I think events used to be in the consensus in TM 0.33 or something like that, but it was later removed (it's up to ABCI apps to do it themselves now)

**Comment by yihuang** [comment]
> I think events used to be in the consensus in TM 0.33 or something like that, but it was later removed (it's up to ABCI apps to do it themselves now)

that's great, @mofhusseini has run a node with this version, haven't seen app mismatch issue yet.

**Comment by tomtau** [comment]
Just for reference: events were perhaps hashed into block headers pre-0.33, not hashed in 0.33, but considered to be hashed again for 0.34, but it was decided not to hash them: https://github.com/tendermint/tendermint/pull/5134#issuecomment-663698684 :D 
I believe it's still the case and even 0.35 remains like this, but I didn't follow the development in detail -- @JayT106 can double-check it.

**Comment by JayT106** [comment]
I think @tomtau you are correct, the [current block header](https://github.com/tendermint/spec/blob/8dd2ed4c6fe12459edeb9b783bdaaaeb590ec15c/spec/core/data_structures.md#header) doesn't have the event hash in v0.34 and later release.

Only `LastResultsHash` has been stored in the header to represent the hash of the transactions' results

**Pull request merged**